src/server/game/Entities/Object/Object.cpp

```diff
@@ -1141,6 +1143,11 @@ bool WorldObject::_IsWithinDist(WorldObject const* obj, float dist2compare, bool
     return distsq < maxdist * maxdist;
 }
 
+bool WorldObject::_IsWithinDist(WorldObject const* obj, float dist2compare, bool is3D) const
+{
+    return _IsWithinDist(obj, dist2compare, is3D, true, true);
```

Comment:
isn't this the same as https://github.com/TrinityCore/TrinityCore/pull/16290/files#diff-29afb15ead2a8204cac07d94f074f260R1114 with the last 2 arguments are optional ?